### ISSUE TITLE
Feature Request: allow opening remote workspaces through extensions

### ISSUE THREAD

**Issue by christian-bromann** [microsoft/vscode-remote-release#6243]
Hey 👋 

I am developing a home screen extension that helps to manage different workspaces. One of my requirements is to allow opening remote workspaces, e.g. codespaces, remote machines etc.. Unfortunately there is not documentation about that and it appears to me that `vscode.openFolder` is not capable doing that. So far I know:

- `vscode.env.remoteName` gives me the environment type, e.g. codespaces, which allows me to determine whether the workspace is local or remote
- while searching this repo for more information I found that there is something like `remote-containers.openFolder` which is not documented though nor do the examples I found make any sense to me
- I know VSCode is able to open codespaces through a like like `vscode://github.codespaces/connect?name=<codespace-id>` but creating a `vscode.Uri` in that form doesn't work

Any help on that matter is appreciated.
Thanks!

**Comment by chrmarti** [comment]
There is currently no API (`remote-containers.openFolder` is used in the UI) for opening a dev container with Remote-Containers.

Not sure about Codespaces. @osortega Are the `vscode://github.codespaces/connect` URIs documented API?

@christian-bromann Do you need a way to open both Remote-Containers and Codespaces dev containers or are you mainly interested in Codespaces? Is the home screen extension you mention a VS Code extension?

**Comment by christian-bromann** [comment]
@chrmarti thanks for the reply.

I need a way to open all possible workspaces, local and remote ones. The [home screen VSCode extension](https://marquee-seven.vercel.app/) extension has a widget that displays all workspaces a user has been working on and allows to easily switch back and forth. So ideally I am looking for a way to open all kind of projects, local or remote ones (run in a remote container or codespaces).

**Comment by osortega** [comment]
I don't think we have those URIs documented anywhere but our protocol handler should be able to pick up a URI with the following structure as long as the extension is installed.
`vscode://github.codespaces/connect?name=${encodeURIComponent(<codespace-name>)}`

Hope this works

**Comment by christian-bromann** [comment]
@osortega how would I open such a codespace? I tried with the [`openFolder`](https://code.visualstudio.com/api/references/commands) command with the exact same `vscode.Uri` without success.

**Comment by osortega** [comment]
Can you try with 
```
const uri = vscode.Uri.parse(`vscode://github.codespaces/connect?name=${encodeURIComponent(<codespace-name>)}`);
vscode.env.openExternal(uri);
```
